### PR TITLE
Update Australia holidays

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Update Australia holidays (according to official sources).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

**Review by arkid15r — APPROVED** [review]
LGTM, great work on 🇦🇺 

Just a couple of suggestions:

File: holidays/countries/australia.py [hunk 1]
```diff
@@ -46,95 +79,112 @@ def __init__(self, *args, **kwargs):
...
+        # from 2015: SAT, SUN - add MON.
 
-        # Labour Day
-        self._add_holiday_1st_mon_of_oct("Labour Day")
+        # New Year's Day.
+        name = "New Year's Day"
+        if self._year >= 2015:
+            self._add_observed(self._add_new_years_day(name))
+        elif self._year >= 1959:
+            dt = (JAN, 1)
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/australia.py]: Done.

**Review by arkid15r — APPROVED** [review]
Merging this in, thanks @KJhellico

**Comment by engserv** [comment]
Kings birthday is on September 23rd in Western Australia (WA) this in 2024.
https://www.wa.gov.au/service/employment/workplace-agreements/public-holidays-western-australia

**Comment by arkid15r** [comment]
> Kings birthday is on September 23rd in Western Australia (WA) in 2024. https://www.wa.gov.au/service/employment/workplace-agreements/public-holidays-western-australia

Thanks for reporting this, I've created a new [issue](https://github.com/vacanza/python-holidays/issues/1756) based on your comment.

**Pull request merged**